examples/hyperparameter_tuning/README.md

```diff
@@ -0,0 +1,202 @@
+# Hyperparameter Tuning Pipeline
```

Comment:
I would name this examples folder name `dynamic_pipelines` and the title of this README to be 'Doing a grid search based Hyper-parameter tuning with Dynamic Pipelines' rather than hyper-parameter tuning. The reason is that this is a limited HPO approach and most users might expect more sophisticated mechanisms when they think of hyper-parameter tuning.

Reply:
I already have a sub-folder named `dyanmic_pipelines` that contains code for the dynamic pipeline logic and is completely independent of the hyperparameter tuning logic.

Reply:
My comment was more to do with the external facing users than the internal structuring. I really get how you did it but I think the name of the example and headings as I mentioned would be more easily consumable when titled as proposed